src/core.ts

```diff
@@ -208,6 +232,63 @@ function mapContracts(solcOutput: SolcOutput): ContractMap {
   return res;
 }
 
+function getInternalVariables(contract: ContractDefinition, contractMap: ContractMap): VariableDeclaration[] {
+  const parents = contract.linearizedBaseContracts.map(id => mustGet(contractMap, id));
+
+  const overriden = new Set<number>();
+  const res = [];
+
+  for (const parent of parents) {
+    for (const v of findAll('VariableDeclaration', parent)) {
+      if (v.stateVariable && v.visibility === 'internal' && !overriden.has(v.id)) {
+        res.push(v);
+        overriden.add(v.id);
+      }
+    }
+  }
+
+  return res;
+}
+
+function getGetterTypes(v: VariableDeclaration): Argument[] {
+  if (!v.typeName) {
+    throw new Error('missing typenName');
+  }
+  return _getGetterTypes(v.typeName, []);
+}
+
+function _getGetterTypes(v: TypeName, types: Argument[]): Argument[] {
```

Comment:
Can you inline this function into `getGetterTypes`? The same for `_getMappingValues` with `getGetterValues`.